src/tests/timeline.rs

```diff
@@ -0,0 +1,147 @@
+use crate::{
+    models::{
+        slot::Slot,
+        timeline::{Timeline, TimelineOperations},
+    },
+    tests::utils::*,
+};
+use chrono::NaiveDate;
+use std::collections::BTreeSet;
+
+#[test]
+fn test_initialize() {
+    let sample_slot = get_slot_1();
+
+    let expected_slot_in_timeline = Slot {
+        start: sample_slot.start,
+        end: sample_slot.end,
+    };
+    let mut expected_collection_in_timeline = BTreeSet::new();
+    expected_collection_in_timeline.insert(expected_slot_in_timeline);
+    let expected_timeline = Timeline {
+        slots: expected_collection_in_timeline,
+    };
+
+    if let Some(timeline) = Timeline::initialize(sample_slot.start, sample_slot.end) {
+        assert_eq!(expected_timeline, timeline);
+    } else {
+        assert!(false);
+    }
+}
+
+#[test]
+fn test_remove_from() {
+    // sample_slot: [2022-10-01 05:00:00 --- 2022-10-01 20:00:00]
```

Comment:
It is important to know the return values specially this function return a static data.
But anyways this is temporary, because I will refactor these functions into flexible functions which accepting params